rest/route/project.go

```diff
@@ -474,22 +473,6 @@ func (h *projectIDPatchHandler) Run(ctx context.Context) gimlet.Responder {
 		}
 	}
 
-	// TODO (PM-2950): remove this temporary conditional initialization for the
-	// vault once the AWS infrastructure is productionized and AWS admin
-	// settings are set.
-	if h.vault == nil && (len(h.apiNewProjectRef.DeleteContainerSecrets) != 0 || len(h.apiNewProjectRef.ContainerSecrets) != 0) {
```

Comment:
The vault still needs to be initialized so that it can connect to Secrets Manager for the later functions.

I would remove the `vault` field from the route handler and instead just initialize the vault conditionally when `len(h.apiNewProjectRef.DeleteContainerSecrets) != 0 || len(h.apiNewProjectRef.ContainerSecrets) != 0`, then pass it into the relevant functions later in the functions:
```go
var vault cocoa.Vault
if len(h.apiNewProjectRef.DeleteContainerSecrets) != 0 || len(h.apiNewProjectRef.ContainerSecrets) != 0 {
    v, err := ...
    vault = v
}
// More stuff...
data.DeleteContainerSecrets(ctx, vault, ...)
// More stuff...
data.UpsertContainerSecrets(ctx, vault, ...)
```